@xen-orchestra/lite/src/components/menu/AppMenu.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <slot :is-open="isOpen" :open="open" name="trigger" />
+  <Teleport to="body" :disabled="!isRoot || !slots.trigger">
+    <ul
+      v-if="!$slots.trigger || isOpen"
+      ref="menu"
+      :class="{ horizontal, shadow }"
+      class="app-menu"
+      v-bind="$attrs"
+    >
+      <slot />
+    </ul>
+  </Teleport>
+</template>
+
+<script lang="ts" setup>
+import placement, { type Options } from "placement.js";
+import { inject, nextTick, provide, ref, toRef, unref, useSlots } from "vue";
+import { onClickOutside, unrefElement, whenever } from "@vueuse/core";
+
+const props = defineProps<{
+  horizontal?: boolean;
+  shadow?: boolean;
+  disabled?: boolean;
+  placement?: Options["placement"];
+}>();
+const isRoot = inject("isMenuRoot", true);
+provide("isMenuRoot", false);
+const slots = useSlots();
+const isOpen = ref(false);
+const menu = ref();
+const isParentHorizontal = inject("isMenuHorizontal", undefined);
+provide("isMenuHorizontal", toRef(props, "horizontal"));
+provide("isMenuDisabled", toRef(props, "disabled"));
+let clearClickOutsideEvent: (() => void) | undefined;
+
+whenever(
+  () => !isOpen.value,
+  () => clearClickOutsideEvent?.()
+);
+
+if (slots.trigger && !inject("closeMenu", false)) {
+  provide("closeMenu", () => (isOpen.value = false));
+}
+
+const open = (event: MouseEvent) => {
+  if (isOpen.value) {
+    return (isOpen.value = false);
+  }
+
+  isOpen.value = true;
```

Comment:
Should we be able to open the menu even if it's disabled?

Reply:
I think so.

I don't think it would be problematic to see the submenu of a disabled item (since all submenu items would be disabled too).

I just checked a desktop app I'm using, the submenu is accessible even if none of its items are.

The only difference is that the parent item is not displayed as "disabled".

So I propose either to keep the current behavior or showing the parent item as active.

Reply:
I made more tests on some desktop apps.

To mimic the general behavior, the item which opens a submenu should always be active. What do you think?

Reply:
I think it would be better to show the parent as active then. I don't think we should ever be able to interact with a disabled element.

Reply:
I agree 👍🏻